stl/inc/xutility

```diff
@@ -4378,6 +4378,28 @@ _OutIt _Copy_memmove(move_iterator<_InIt> _First, move_iterator<_InIt> _Last, _O
     return _Copy_memmove(_First.base(), _Last.base(), _Dest);
 }
 
+template <class _InIt, class _OutIt>
+_OutIt _Copy_memmove_common(_InIt _IFirst, _InIt _ILast, _OutIt _OFirst, _OutIt _OLast) noexcept {
+    const char* const _IFirst_ch = const_cast<const char*>(reinterpret_cast<const volatile char*>(_IFirst));
+    const char* const _ILast_ch  = const_cast<const char*>(reinterpret_cast<const volatile char*>(_ILast));
+    char* const _OFirst_ch       = const_cast<char*>(reinterpret_cast<volatile char*>(_OFirst));
+    const char* const _OLast_ch  = const_cast<const char*>(reinterpret_cast<const volatile char*>(_OLast));
+    const auto _Count            = static_cast<size_t>((_STD min)(_ILast_ch - _IFirst_ch, _OLast_ch - _OFirst_ch));

```

Comment:
In new code, we generally follow a convention of using `auto` when the RHS is a cast - the rationale being that the type information is right there, so repeating it serves no purpose. (Old code hasn't been completely modernized to follow this, IIRC, creating inconsistent precedent - sorry!) So I recommend:
```suggestion
    const auto _IFirst_ch = const_cast<const char*>(reinterpret_cast<const volatile char*>(_IFirst));
    const auto _ILast_ch  = const_cast<const char*>(reinterpret_cast<const volatile char*>(_ILast));
    const auto _OFirst_ch = const_cast<char*>(reinterpret_cast<volatile char*>(_OFirst));
    const auto _OLast_ch  = const_cast<const char*>(reinterpret_cast<const volatile char*>(_OLast));
    const auto _Count     = static_cast<size_t>((_STD min)(_ILast_ch - _IFirst_ch, _OLast_ch - _OFirst_ch));
```